tests/integration/conftest.py

```diff
@@ -0,0 +1,128 @@
+    def __init__(self):
+        self.session = requests.Session()
+        self._token = None
+        self.csrf_token = None
+
+    def request(self, endpoint, method='GET', headers=None, auth='default',
+                expected_code=200, **kwargs):
+        headers = dict(headers or {})
+        if self.csrf_token:
+            headers['X-CSRFToken'] = self.csrf_token
+        if auth == 'token':
+            headers['Authorization'] = 'token {}'.format(self.get_token())
+        elif auth == 'default':
+            if OPENLCS_LOCAL_LOGIN and 'sessionid' not in self.session.cookies:
+                # Obtain CSRF token
+                self.request('/', auth=None)
+                self.request(
+                    '/login/',
+                    method='POST',
+                    data=OPENLCS_LOGIN_DATA,
+                    auth=None,
+                )
+            else:
+                kwargs['auth'] = HTTPKerberosAuth(mutual_authentication=False)
+        elif auth is None:
+            # No auth
+            pass
+        else:
+            assert False, "Unknown auth type"
+        if not urlparse(endpoint).netloc:
+            endpoint = '{}/{}'.format(OPENLCS_URL, endpoint.lstrip('/'))
+        self.assert_url_same_host(endpoint)
+        response = self.session.request(
+            method,
+            endpoint,
+            headers=headers,
+            **kwargs
+        )
+        self.csrf_token = self.session.cookies.get(
+            'csrftoken',
+            default=self.csrf_token,
+        )
+        # requests internally add some headers, we don't want them in curlify
+        # output
+        response.request.headers = headers
+        print(self.to_curl(response.request))
+        assert not expected_code or response.status_code == expected_code, (
+            "Request returned status {}, expected {}\nResponse body:\n{}"
+            .format(response.status_code, expected_code, response.content)
+        )
+        return response
+
+    def to_curl(self, request):
+        if request.body and request.body.startswith('--'):
+            # It would choke on binary input
+            request.body = 'XXX Multipart content omitted'
+        try:
+            import curlify
+            return curlify.to_curl(request)
+        except ImportError:
+            return "To see curl command invocation, please install curlify"
+
+    def api_call(self, endpoint, method='GET', headers=None, return_json=True,
+                 expected_code=200, **kwargs):
+        if not urlparse(endpoint).netloc:
+            endpoint = 'rest/v1/{}'.format(endpoint.lstrip('/'))
+        headers = dict(headers or {})
+        headers['Accept'] = 'application/json'
+        response = self.request(
+            endpoint=endpoint,
+            method=method,
+            headers=headers,
+            auth='token',
+            expected_code=expected_code,
+            **kwargs
+        )
+        if not return_json:
```

Comment:
`return_json` is `False`, then need a return?

Reply:
Updated.